File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ProtocolDiagnostic.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class ProtocolDiagnostic(
  val location: ProtocolLocation,
  val message: String,
  val severity: DiagnosticSeverity, // Oneof: error, warning, info, suggestion
  val code: String? = null,
  val source: String? = null,
  val relatedInformation: List<ProtocolRelatedInformationDiagnostic>? = null,
)

File: agent/bindings/csharp/ProtocolDiagnostic.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ProtocolDiagnostic
  {
    [JsonProperty(PropertyName = "location")]
    public ProtocolLocation Location { get; set; }
    [JsonProperty(PropertyName = "message")]
    public string Message { get; set; }
    [JsonProperty(PropertyName = "severity")]
    public DiagnosticSeverity Severity { get; set; } // Oneof: error, warning, info, suggestion
    [JsonProperty(PropertyName = "code")]
    public string Code { get; set; }
    [JsonProperty(PropertyName = "source")]
    public string Source { get; set; }
    [JsonProperty(PropertyName = "relatedInformation")]
    public ProtocolRelatedInformationDiagnostic[] RelatedInformation { get; set; }
  }
}
